mkosi/util.py

```diff
@@ -77,8 +77,13 @@ def flatten(lists: Iterable[Iterable[T]]) -> list[T]:
 class INVOKING_USER:
     uid = int(os.getenv("SUDO_UID") or os.getenv("PKEXEC_UID") or os.getuid())
     gid = int(os.getenv("SUDO_GID") or os.getgid())
-    name = pwd.getpwuid(uid).pw_name
-    home = Path(f"~{name}").expanduser()
+    # names for UIDs might not be always resolvable, e.g. during reproducible builds
```

Comment:
Hmm, I have to say that the comment doesn't make sense to me. Why would it matter whether a build is reproducible or not matter for name resolution? Do you maybe mean that a repeat build that does not set up the environment properly hits this case?

Isn't https://github.com/systemd/mkosi/issues/2081 really just a problem with how the environment is set up? The uid we are running under should have a name. If it doesn't many things will complain. If somebody is trying to repeat a build, they must take care of /etc/passwd and /etc/group too.

Reply:
This is probably too short, yes, but I'd like to keep the context in there. How about:
> names for UIDs might not be always resolvable, e.g. during build time without proper NSS, which is used during reproducible builds so as not to leak the username of the build user